hedera-node/cli-clients/src/main/java/com/hedera/services/cli/sign/RecordStreamSigningUtils.java

```diff
@@ -0,0 +1,287 @@
+
+        Objects.requireNonNull(signatureFileDestination, "signatureFileDestination must not be null");
+        Objects.requireNonNull(streamFileToSign, "streamFileToSign must not be null");
+        Objects.requireNonNull(keyPair, "keyPair must not be null");
+        Objects.requireNonNull(hapiVersion, "hapiVersion must not be null");
+
+        try {
+            final var recordFile = streamFileToSign.toFile();
+            final int version = getRecordStreamVersion(recordFile);
+            if (version != SUPPORTED_STREAM_FILE_VERSION) {
+                System.err.printf(
+                        "signRecordStreamFile :: Failed to sign file [%s] with unsupported version [%s]%n",
+                        streamFileToSign.getFileName(), version);
+                return false;
+            }
+
+            initRecordDigest();
+            int[] fileHeader = createFileHeader(hapiVersion);
+
+            outputStreamDigest(fileHeader, recordFile.getAbsolutePath());
+            SignatureFile.Builder signatureFile = createSignatureFile(keyPair);
+            generateSigRecordStreamFile(signatureFileDestination.toFile(), signatureFile);
+
+            System.out.println("Generated signature file: " + signatureFileDestination);
+
+            return true;
+        } catch (final SignatureException | InvalidProtobufVersionException | IOException e) {
+            System.err.printf(
+                    "signRecordStreamFile :: Failed to sign file [%s] with exception : [%s]%n",
+                    streamFileToSign.getFileName(), e);
+            return false;
+        } catch (final InvalidKeyException | NoSuchAlgorithmException | NoSuchProviderException e) {
+            System.err.printf("signRecordStreamFile :: Irrecoverable error encountered :[%s]%n", e);
+            throw new RuntimeException("Irrecoverable error encountered", e);
+        }
+    }
+
+    private static int getRecordStreamVersion(File recordFile) throws IOException {
+        try {
+            final var isCompressed = RecordStreamType.getInstance().isGzFile(recordFile.getName());
+            if (isCompressed) {
+                final var uncompressedFileContents =
+                        FileCompressionUtils.readUncompressedFileBytes(recordFile.getAbsolutePath());
+                return ByteBuffer.wrap(uncompressedFileContents, 0, 4).getInt();
+            } else {
+                return readFirstIntFromFile(recordFile);
+            }
+
+        } catch (final IOException e) {
+            System.err.printf(
+                    "getRecordStreamVersion :: Failed to read record stream version from file [%s] with exception : [%s]%n",
+                    recordFile.getName(), e);
+            throw e;
+        }
+    }
+
+    private static int[] createFileHeader(@NonNull final String hapiVersion) throws InvalidProtobufVersionException {
+        Objects.requireNonNull(hapiVersion, "hapiVersion must not be null");
+
+        //  example: "0.37.0-allowance-SNAPSHOT";
```

Comment:
There's an example of the _input_ of this following statement.  But what's an example of the expected _output_?  Also, this seems "finicky" enough that it should probably be a separate little method with its own little test.

Actually I'm a little surprised that this thing - pull major/minor/patch out of a version string, isn't a utility method _somewhere_ - probably in some library this repo uses?

Reply:
Removed it. It was used for testing. This is refactored code.